realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -183,10 +184,10 @@ public static synchronized void init(Context context) {
             if (context == null) {
                 throw new IllegalArgumentException("Non-null context required.");
             }
-            BaseRealm.applicationContext = context.getApplicationContext();
-            RealmCore.loadLibrary(BaseRealm.applicationContext);
+            RealmCore.loadLibrary(context);
             defaultConfiguration = new RealmConfiguration.Builder().build();
             RealmLog.add(BuildConfig.DEBUG ? new AndroidLogger(Log.DEBUG) : new AndroidLogger(Log.WARN));
+            BaseRealm.applicationContext = context.getApplicationContext();
```

Comment:
I don't think so? `context` is the input to the method and there is a nullcheck just above here?